File: 38.java
class Solution {
    public String countAndSay(int n) {
        String res = "1";
        String tempRes = "";
        for (int i = 0; i < n-1; i++) {
            tempRes = "";
            int count = 0;
            char currChar = res.charAt(0);
            for (int j = 0; j < res.length(); j++) {
                if (currChar == res.charAt(j)) count += 1;
                else {
                    tempRes += count + "" + currChar;
                    currChar = res.charAt(j);
                    count = 1;
                }
            }
            tempRes += count + "" + currChar;
            res = tempRes;
        }
        return res;
    }
}

File: 38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        res = '1'
        for i in range(n-1):
            new_char = res[0]
            temp_res = ''
            count = 0
            for char in res:
                if new_char == char: count += 1
                else:
                    temp_res += (str(count) + new_char)
                    new_char = char
                    count = 1
            res = temp_res + str(count) + char
        return res
